crates/server/src/authority/catalog.rs

```diff
@@ -430,25 +430,40 @@ async fn lookup<'a, R: ResponseHandler + Unpin>(
     response_edns: Option<Edns>,
     response_handle: R,
 ) -> LookupControlFlow<ResponseInfo> {
+    let request_header = request.header();
+    let request_id = request.id();
+    let edns = request.edns();
+    let origin = authority.origin();
```

Comment:
Nit: I'd suggest to (1) inline these for bindings that are used only once and (2) defining them as late as possible to improve code locality.

Reply:
This should be resolved.